nilearn/glm/first_level/first_level.py

```diff
@@ -710,8 +716,24 @@ def fit(
                    :obj:`list` or :obj:`tuple` of Niimg-like objects, \
                    SurfaceImage object, \
                    or :obj:`list` or :obj:`tuple` of SurfaceImage
-            Data on which the :term:`GLM` will be fitted. If this is a list,
-            the affine is considered the same for all.
+            Data on which the :term:`GLM` will be fitted.
+            If this is a list, the affine is considered the same for all.
+
+            .. warning::
+
+                If the FirstLevelModel object was instantiated
+                with a ``mask_img``,
+                then ``run_imgs`` must be compatible with ``mask_img``.
+                For example, if ``mask_img`` is
+                a :class:`nilearn.maskers.NiftiMasker` instance
+                or a Niimng-like object, then ``run_imgs`` must be a
+                Niimg-like object, \
+                a :obj:`list` or a :obj:`tuple` of Niimg-like objects.
+                If ``mask_img`` is
+                a ``SurfaceMasker`` or ``SurfaceImage`` instance,
+                then ``run_imgs`` mustr be a
+                ``SurfaceImage`` object, \
+                a :obj:`list` or a :obj:`tuple` of ``SurfaceImage`` objects.
```

Comment:
Putting this in the doc string for now but this will probably need to be added to a few other places.